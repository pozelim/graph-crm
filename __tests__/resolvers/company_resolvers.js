/* eslint-disable no-undef */
import { setup, tearDown } from './base';

let request;
let prisma;
let data = {};

async function buildData() {
  const company1 = await prisma.company.create({
    data: { name: 'company1' },
  });

  return {
    company1,
  };
}

afterAll(async () => tearDown(async () => prisma.company.deleteMany({})));

beforeAll(async () => {
  const setupResult = await setup();
  request = setupResult.request;
  prisma = setupResult.prisma;

  await prisma.company.deleteMany({});
  data = await buildData();
});

function create(input) {
  const { name } = input;
  return request
    .post('/graphql')
    .set('Accept', 'application/json')
    .send({
      query: `
      mutation { createCompany(input: {
          name: "${name}"
        }) {
          name
        }
      }`,
    });
}

function getById(id) {
  return request
    .post('/graphql')
    .set('Accept', 'application/json')
    .send({
      query: `
        { company(id: "${id}") {
            name
          }
        }`,
    });
}

it('create company', async () => {
  const response = await create({ name: 'dummy' });
  expect(response.statusCode).toBe(200);
  expect(response.body).toMatchSnapshot();
});

it('get company by id', async () => {
  const response = await getById(data.company1.id);
  expect(response.statusCode).toBe(200);
  expect(response.body).toMatchSnapshot();
});
