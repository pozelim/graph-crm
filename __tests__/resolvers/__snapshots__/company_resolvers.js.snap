// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create company 1`] = `
Object {
  "data": Object {
    "createCompany": Object {
      "name": "dummy",
    },
  },
}
`;

exports[`get company by id 1`] = `
Object {
  "data": Object {
    "company": Object {
      "name": "company1",
    },
  },
}
`;
